test/unit/modules/keyboard.js

```diff
@@ -257,4 +260,245 @@ describe('Keyboard', function () {
+      const quill = this.initialize(Quill, markup, this.container, {
+        modules: {
+          table: true,
+        },
+      });
+      quill.setSelection(18);
+      const keydownEvent = new KeyboardEvent('keydown', {
+        key: 'tab',
+      });
+
+      quill.root.dispatchEvent(keydownEvent);
+      expect(quill.getSelection()).toEqual(new Range(24));
+    });
+
+    it('should select next cell in second row if cursor in the last cell of first row on tab click', function () {
+      const quill = this.initialize(Quill, markup, this.container, {
+        modules: {
+          table: true,
+        },
+      });
+      quill.setSelection(30);
+      const keydownEvent = new KeyboardEvent('keydown', {
+        key: 'tab',
+      });
+
+      quill.root.dispatchEvent(keydownEvent);
+      expect(quill.getSelection()).toEqual(new Range(36));
+    });
+
+    it('should select previous cell on tab + shift click', function () {
+      const quill = this.initialize(Quill, markup, this.container, {
+        modules: {
+          table: true,
+        },
+      });
+      quill.setSelection(31);
+      const keydownEvent = new KeyboardEvent('keydown', {
+        key: 'tab',
+        shiftKey: true,
+      });
+
+      quill.root.dispatchEvent(keydownEvent);
+      expect(quill.getSelection()).toEqual(new Range(29));
+    });
+
+    it('should select previous cell on first row if cursor in the first cell of second row on tab + shift click', function () {
+      const quill = this.initialize(Quill, markup, this.container, {
+        modules: {
+          table: true,
+        },
+      });
+      quill.setSelection(38);
+      const keydownEvent = new KeyboardEvent('keydown', {
+        key: 'tab',
+        shiftKey: true,
+      });
+
+      quill.root.dispatchEvent(keydownEvent);
+      expect(quill.getSelection()).toEqual(new Range(35));
+    });
+
+    it('should select next cell in header on tab click', function () {
+      const quill = this.initialize(Quill, markup, this.container, {
+        modules: {
+          table: true,
+        },
+      });
+      quill.setSelection(0);
+      const keydownEvent = new KeyboardEvent('keydown', {
+        key: 'tab',
+      });
+
+      quill.root.dispatchEvent(keydownEvent);
+      expect(quill.getSelection()).toEqual(new Range(6));
+    });
+
+    it('should select previous cell in header on tab + shift click', function () {
+      const quill = this.initialize(Quill, markup, this.container, {
+        modules: {
+          table: true,
+        },
+      });
+      quill.setSelection(8);
+      const keydownEvent = new KeyboardEvent('keydown', {
+        key: 'tab',
+        shiftKey: true,
+      });
+
+      quill.root.dispatchEvent(keydownEvent);
+      expect(quill.getSelection()).toEqual(new Range(5));
+    });
+  });
+
+  describe('tab navigation on lite table', function () {
+    beforeAll(function () {
+      Quill.register({ 'modules/table': TableLite }, true);
```

Comment:
Cannot this code be executed before other test? In this case all tests will be run for lite table. Can we add restore?

Reply:
fixed